mountpoint-s3-fs/src/metablock/path.rs

```diff
@@ -118,6 +130,23 @@ impl From<ValidKey> for String {
     }
 }
 
+impl TryFrom<String> for ValidKey {
+    type Error = ValidKeyError;
+
+    /// Constructs a valid key performing checks.
+    fn try_from(value: String) -> Result<Self, Self::Error> {
```

Comment:
Can we add tests for this? How does it handle strings ending in "/"?

Also, consider inlining the `validate` method, especially if it helps avoiding the additional scan to get `name_len`.

Reply:
note that `ValidKey::validate` is used in `builder.rs`